pandas/core/indexes/interval.py

```diff
@@ -599,6 +599,14 @@ def closed(self):
         """
         return self._closed
 
+    @property
+    def length(self):
+        """
+        Return an Index with entries denoting the length of each Interval in
+        the IntervalIndex
+        """
+        return self.right - self.left
```

Comment:
Some intervals don't have a length, if values are categorical-like, e.g. letters (``pd.Interval('b', 'e')``). 

I think maybe the length calculation should be wrapped in a try/except statement, with a helpful error message for when length cannot be calculated (e.g. "not possible to calculate length between 'b' and 'e'" in the above example.

Reply:
@topper-123 : I agree with this assessment.  @jschendel : Do wrap the `length` property as such and add tests.

Reply:
Done.  Should probably audit the existing code for more instances of this, as I imagine there are a few places where similar failures occur, e.g. `Interval.mid`.

Reply:
That's probably a good idea.  Depend on how big that change is, you may want to leave that as a follow-up PR though.

Reply:
Yeah, planning to do it in a separate PR so I don't delay this one with unrelated work while I look for more instances.  Cleaner than having the instances I immediately see go in here,  and the others in a separate PR too.